addons/hw_drivers/controllers/driver.py

```
@@ -1,432 +1,379 @@
 
+event_manager = EventManager()
+
+
+#----------------------------------------------------------
+# Manager
+#----------------------------------------------------------
 
+class Manager(Thread):
 
+    def __init__(self):
+        super(Manager, self).__init__()
+        self.load_drivers()
 
+    def load_drivers(self): # Load drivers from IoT Box
+        path = os.path.join(home(), 'odoo/addons/hw_drivers/drivers')
+        driversList = os.listdir(path)
+        for driver in driversList:
+            path_file = os.path.join(path, driver)
+            spec = util.spec_from_file_location(driver, path_file)
+            if spec:
+                module = util.module_from_spec(spec)
+                spec.loader.exec_module(module)
+
+    def send_alldevices(self): # Send device to Odoo
+        server = get_odoo_server_url()
+        if server:
+            iot_box = {'name': socket.gethostname(),'identifier': get_mac_address(), 'ip': get_ip(), 'token': get_token()}
+            devices_list = {}
+            for device in iot_devices:
+                identifier = iot_devices[device].device_identifier
+                devices_list[identifier] = {
+                    'name': iot_devices[device].device_name,
+                    'type': iot_devices[device].device_type,
+                    'connection': iot_devices[device].device_connection,
+                }
+            data = {
+                'params': {
+                    'iot_box' : iot_box,
+                    'devices' : devices_list,
+                }
+            }
+            urllib3.disable_warnings()
+            http = urllib3.PoolManager(cert_reqs='CERT_NONE')
+            try:
+                req = http.request(
+                    'POST',
+                    server + "/iot/setup",
+                    body = json.dumps(data).encode('utf8'),
+                    headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
+                )
+            except:
```

[no bare except !](https://docs.quantifiedcode.com/python-anti-patterns/correctness/no_exception_type_specified.html)